mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/common/ClientChargesTest.java

```diff
@@ -0,0 +1,121 @@
+           
+        //waived off 100 USD charge
+        final String waiveChargeRsponseId=ClientHelper.waiveChargesForClients(this.requestSpec, this.responseSpec,clientId, 
+                   clientChargeId, ClientHelper.getWaiveChargeJSON("100",clientChargeId.toString()));
+           Assert.assertNotNull(waiveChargeRsponseId);
+           
+         //undo client charge transaction
+         final Integer undoWaiveTrxnId =ClientHelper.undoTransaction(this.requestSpec, this.responseSpec,clientId.toString(), 
+                   waiveChargeRsponseId.toString());
+           Assert.assertNotNull(undoWaiveTrxnId);
+           isUndoTransaction(clientId.toString(), undoWaiveTrxnId.toString());
+       
+       //pay charge before client activation date---failure test case
+       final String responseId_activationDate_failure=ClientHelper.payChargesForClients(this.requestSpec,responseSpecFailure,clientId, 
+               clientChargeId, ClientHelper.getPayChargeJSON("30 October 2011", "20"));
+       Assert.assertNull(responseId_activationDate_failure);
+       
+      //pay charge more than outstanding ---failure test case
+       final String responseId_moreAmount_failure=ClientHelper.payChargesForClients(this.requestSpec,responseSpecFailure,clientId, 
+       clientChargeId, ClientHelper.getPayChargeJSON("25 AUGUST 2015", "300"));
+       Assert.assertNull(responseId_moreAmount_failure);
+       
+       //create charge for loan
+       final Integer loanChargeId = ChargesHelper.createCharges(this.requestSpec, this.responseSpec,ChargesHelper.getLoanSpecifiedDueDateJSON());
+       Assert.assertNotNull(loanChargeId);
+       
+       //apply loan charge to client
+       ResponseSpecification responseLoanChargeFailure = new ResponseSpecBuilder().expectStatusCode(403).build();
+       final Integer clientLoanChargeId = ClientHelper.addChargesForClient(this.requestSpec,responseLoanChargeFailure,clientId, 
+               ClientHelper.getSpecifiedDueDateChargesClientAsJSON(loanChargeId.toString(), "29 October 2011"));
+       Assert.assertNull(clientLoanChargeId);
+       
+       //paid 10 USD  charge
+       final String chargePaid_responseId= ClientHelper.payChargesForClients(this.requestSpec, this.responseSpec,clientId, clientChargeId,
+               ClientHelper.getPayChargeJSON("25 AUGUST 2015", "100"));
+       Assert.assertNotNull(chargePaid_responseId); 
+       
+       //validate oustandingAmount
+       isValidOutstandingAmount(ClientHelper.getClientCharge(requestSpec, responseSpec, clientId.toString()), (float)100.0);
+       
+       
+    }
+    //undo transaction
+   private void isUndoTransaction(String clientId,String transactionId){
+       final Boolean isReversed=ClientHelper.getClientTransactions(this.requestSpec, this.responseSpec,clientId.toString(), 
+               transactionId);
+       Assert.assertTrue(isReversed);
+   }
+   
+   private void isValidOutstandingAmount(Object outStandingAmount,Object expectedAmount){
```

Comment:
Files need to be formatted using our formatter , refer https://github.com/openMF/mifosx/blob/develop/Contributing.md point 6